src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -5516,7 +5516,7 @@ protected virtual void Dispose(bool disposing)
                     else
                     {
                         // Since our timeout is in ms and linger is in seconds, implement our own sortof linger here.
-                        errorCode = SocketPal.Shutdown(_handle, SocketShutdown.Send);
+                        errorCode = SocketPal.Shutdown(_handle, true, SocketShutdown.Send);
```

Comment:
Just my personal preference, but consider naming the arguments so that it's clear at the call site what "true" here means.